### ISSUE TITLE
Error Server 500 Acunetix Parser

### ISSUE THREAD

**Issue by zfirered** [DefectDojo/django-DefectDojo#1936]
**Bug description**
Error while upload acunetix xml file
this is the sample when i'am upload it and the server error 500 happen

**Deployment method** *(select with an `X`)*
- [ ] Kubernetes
- [X] Docker
- [ ] setup.bash / legacy-setup.bash

**Sample scan files** (optional)
[20200228_XML_http_www_itsecgames_com_.zip](https://github.com/DefectDojo/django-DefectDojo/files/4265131/20200228_XML_http_www_itsecgames_com_.zip)

**Comment by valentijnscholten** [comment]
do you have logs? which version/branch are you running?

**Comment by zfirered** [comment]
all version has this bud and not yet fixed

**Comment by valentijnscholten** [comment]
What error do you see in the logs? In the provided sample file every tag has it's data inside a CDATA element, not sure if this is normal?

**Comment by zfirered** [comment]
![image](https://user-images.githubusercontent.com/17750324/75770058-d093e200-5d79-11ea-8177-5f58e1609e76.png)

**Comment by zfirered** [comment]
![dojo import error](https://user-images.githubusercontent.com/17750324/75770146-033dda80-5d7a-11ea-85e3-bdb78bb03f1e.png)

**Comment by zfirered** [comment]
if acunetix found this, Slow HTTP Denial of Service Attack, then dojo won't import properly always Error Server 500

**Comment by zfirered** [comment]
<ReportItem id="78" color="orange">
                <Name><![CDATA[Slow HTTP Denial of Service Attack]]></Name>
                <ModuleName><![CDATA[slow-loris]]></ModuleName>
                <Details><![CDATA[Time difference between connections: <code>83860937 </code>ms]]></Details>
                <Affects><![CDATA[/]]></Affects>
                <Parameter><![CDATA[]]></Parameter>
                <AOP_SourceFile><![CDATA[]]></AOP_SourceFile>
                <AOP_SourceLine></AOP_SourceLine>
                <AOP_Additional><![CDATA[]]></AOP_Additional>
                <IsFalsePositive><![CDATA[]]></IsFalsePositive>
                <Severity><![CDATA[medium]]></Severity>
                <Type><![CDATA[denialofservice]]></Type>
                <Impact><![CDATA[A single machine can take down another machine's web server with minimal bandwidth and side effects on unrelated services and ports.]]></Impact>
                <Description><![CDATA[Your web server is vulnerable to Slow HTTP DoS (Denial of Service) attacks. <br/><br/>
Slowloris and Slow HTTP POST DoS attacks rely on the fact that the HTTP protocol, by design, requires requests to be completely received by the server before they are processed. If an HTTP request is not complete, or if the transfer rate is very low, the server keeps its resources busy waiting for the rest of the data. If the server keeps too many resources busy, this creates a denial of service.]]></Description>
                <DetailedInformation><![CDATA[]]></DetailedInformation>
                <Recommendation><![CDATA[Consult Web references for information about protecting your web server against this type of attack.]]></Recommendation>
                <TechnicalDetails>
                    <Request><![CDATA[]]></Request>
                </TechnicalDetails>
                <CWEList>
                    
                </CWEList>
                <CVEList>
                    
                </CVEList>
                <CVSS>
                  <Descriptor><![CDATA[AV:N/AC:L/Au:N/C:N/I:N/A:P]]></Descriptor>
                  <Score><![CDATA[5.0]]></Score>
                  <AV><![CDATA[Network_Accessible]]></AV>
                  <AC><![CDATA[Low]]></AC>
                  <Au><![CDATA[None]]></Au>
                  <C><![CDATA[None]]></C>
                  <I><![CDATA[None]]></I>
                  <A><![CDATA[Partial]]></A>
                  <E><![CDATA[]]></E>
                  <RL><![CDATA[]]></RL>
                  <RC><![CDATA[]]></RC>
                </CVSS>
                
                <CVSS3>
                  <Descriptor><![CDATA[CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L]]></Descriptor>
                  <Score><![CDATA[5.3]]></Score>
                  <TempScore><![CDATA[]]></TempScore>
                  <EnvScore><![CDATA[]]></EnvScore>
                  <AV><![CDATA[Network]]></AV>
                  <AC><![CDATA[Low]]></AC>
                  <PR><![CDATA[None]]></PR>
                  <UI><![CDATA[None]]></UI>
                  <S><![CDATA[Unchanged]]></S>
                  <C><![CDATA[None]]></C>
                  <I><![CDATA[None]]></I>
                  <A><![CDATA[Low]]></A>
                  <E><![CDATA[]]></E>
                  <RL><![CDATA[]]></RL>
                  <RC><![CDATA[]]></RC>
                </CVSS3>
                
                <References>
                    
                    <Reference>
                      <Database><![CDATA[Slowloris DOS Mitigation Guide]]></Database>
                      <URL><![CDATA[https://www.funtoo.org/Slowloris_DOS_Mitigation_Guide]]></URL>
                    </Reference>
                    
                    <Reference>
                      <Database><![CDATA[Protect Apache Against Slowloris Attack]]></Database>
                      <URL><![CDATA[https://web.archive.org/web/20180329210925/http://blog.secaserver.com/2011/08/protect-apache-slowloris-attack/]]></URL>
                    </Reference>
                    
                </References>
            </ReportItem>

**Comment by jrsfaisal** [comment]
I got some issues about acunetix. I'm try import XML from acunetix 10.5, but i got 500 Error. This is for docker log.

```
nginx_1         | 2020/03/04 16:41:33 [warn] 6#6: *2 a client request body is buffered to a temporary file /var/cache/nginx/client_temp/0000000004, client: 172.18.0.1, server: , request: "POST /product/8/import_scan_results HTTP/1.1", host: "localhost:8080", referrer: "http://localhost:8080/product/8/import_scan_results"
uwsgi_1         | Internal Server Error: /product/8/import_scan_results
uwsgi_1         | Traceback (most recent call last):
uwsgi_1         |   File "/usr/local/lib/python3.5/site-packages/django/core/handlers/exception.py", line 34, in inner
uwsgi_1         |     response = get_response(request)
uwsgi_1         |   File "/usr/local/lib/python3.5/site-packages/django/core/handlers/base.py", line 115, in _get_response
uwsgi_1         |     response = self.process_exception_by_middleware(e, request)
uwsgi_1         |   File "/usr/local/lib/python3.5/site-packages/django/core/handlers/base.py", line 113, in _get_response
uwsgi_1         |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
uwsgi_1         |   File "/usr/local/lib/python3.5/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
uwsgi_1         |     return view_func(request, *args, **kwargs)
uwsgi_1         |   File "./dojo/product/views.py", line 413, in import_scan_results_prod
uwsgi_1         |     return import_scan_results(request, pid=pid)
uwsgi_1         |   File "/usr/local/lib/python3.5/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
uwsgi_1         |     return view_func(request, *args, **kwargs)
uwsgi_1         |   File "./dojo/engagement/views.py", line 523, in import_scan_results
uwsgi_1         |     parser = import_parser_factory(file, t, active, verified)
uwsgi_1         |   File "./dojo/tools/factory.py", line 142, in import_parser_factory
uwsgi_1         |     parser = AcunetixScannerParser(file, test)
uwsgi_1         |   File "./dojo/tools/acunetix/parser.py", line 17, in __init__
uwsgi_1         |     acunetix_defectdojo_findings = get_defectdojo_findings(xml_output)
uwsgi_1         |   File "./dojo/tools/acunetix/parser_helper.py", line 175, in get_defectdojo_findings
uwsgi_1         |     cwe = report_item['CWEId']
uwsgi_1         | KeyError: 'CWEId'
uwsgi_1         | [pid: 1|app: 0|req: 24/24] 172.18.0.1 () {60 vars in 1365 bytes} [Wed Mar  4 16:41:33 2020] POST /product/8/import_scan_results => generated 27 bytes in 77 msecs (HTTP/1.1 500) 4 headers in 126 bytes (1 switches on core 0)
nginx_1         | 172.18.0.1 - - [04/Mar/2020:16:41:33 +0000] "POST /product/8/import_scan_results HTTP/1.1" 500 27 "http://localhost:8080/product/8/import_scan_results" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.122 Safari/537.36" "-"
```

**Comment by zfirered** [comment]
is there any update regarding this one?

**Comment by zfirered** [comment]
gaesss??

**Comment by valentijnscholten** [comment]
> which version/branch are you running?

**Comment by zfirered** [comment]
> > which version/branch are you running?
> 
> exact commit needed

ALL VERSION has this problem... i test it...

**Comment by valentijnscholten** [comment]
The reason why we're asking is because a stacktrace shows filenames and linenumbers and it's much easier for us to find out what is going on when we have the exact commit that goed with the stacktrace (which you provded as a screenshot).
We are volunteers and ask you to provide some basic information. Even if it might not make a big change in this particular case, you probably get a lot less people interested in looking at if you only provide half of the information

**Comment by zfirered** [comment]
well im sorry for that, but all version had this issue, you can try your self, i have attachment

**Comment by steeve85** [comment]
I opened a PR that should fix this issue https://github.com/DefectDojo/django-DefectDojo/pull/2185

**Comment by valentijnscholten** [comment]
Should be fixed, please reopen when needed.

**Comment by zfirered** [comment]
hey thanks for fix!

here another problem
![image](https://user-images.githubusercontent.com/17750324/80029795-7c071a80-8511-11ea-87b4-e7fca91a8f0f.png)

Line 9476 is:
<Details><![CDATA[  URL encoded POST input <strong><span class="bb-dark">emails</span></strong> was set to <strong><span class="bb-dark">1 À§À¢%2527%2522</span></strong><br/><br/>  Error message found: <pre><span class="bb-blue">A Database Error Occurred</span></pre>  ]]></Details>

**Comment by zfirered** [comment]
also can you make more detail for acunetix?
example if i got findings about directory listing 3 list so you can add actualy 3 list not count as 1 finding

**Comment by zfirered** [comment]
> Should be fixed, please reopen when needed.

Please Re Open

**Comment by steeve85** [comment]
@zfirered Can you please Open a new issue with the error details and the file you are trying to import (if possible)? I will take a look to fix it.
If the file contains some private data, please provide a sample file with the similar "email" that breaks the parser.
Thanks

**Comment by LeonardoNve** [comment]
Same problem here, last version of DD (cloned today)
[20200504_XML_192_168_0_1.zip](https://github.com/DefectDojo/django-DefectDojo/files/4577253/20200504_XML_192_168_0_1.zip)

:

![image](https://user-images.githubusercontent.com/6109592/81010423-95d02800-8e56-11ea-9813-10714898f59a.png)

An example of the acunetix file I´m uploading attached.

**Comment by zfirered** [comment]
> Same problem here, last version of DD (cloned today)
> [20200504_XML_192_168_0_1.zip](https://github.com/DefectDojo/django-DefectDojo/files/4577253/20200504_XML_192_168_0_1.zip)
> 
> :
> 
> ![image](https://user-images.githubusercontent.com/6109592/81010423-95d02800-8e56-11ea-9813-10714898f59a.png)
> 
> An example of the acunetix file I´m uploading attached.

Which version do you have? it's already Fixed
Slow HTTP Denial of Service Attack the problem, i saw from your file

**Comment by zfirered** [comment]
> Same problem here, last version of DD (cloned today)
> [20200504_XML_192_168_0_1.zip](https://github.com/DefectDojo/django-DefectDojo/files/4577253/20200504_XML_192_168_0_1.zip)
> 
> :
> 
> ![image](https://user-images.githubusercontent.com/6109592/81010423-95d02800-8e56-11ea-9813-10714898f59a.png)
> 
> An example of the acunetix file I´m uploading attached.

Which version do you have? i'm able to import this without any problem
![image](https://user-images.githubusercontent.com/17750324/81024379-76a8b900-8e9d-11ea-8c7c-f779e3e3b7aa.png)

**Comment by LeonardoNve** [comment]
@zfirered 

Version: v. 1.5.4 ( release mode )
Installed on Kali linux 2019.4 with Docker deployment. Installed from github yesterday some minutes before my post.

Thanks.

**Comment by valentijnscholten** [comment]
The fixes are only on the dev branch and 1.6 branch
